impl/src/main/java/org/jboss/weld/bootstrap/WeldStartup.java

```diff
@@ -213,6 +214,11 @@ public WeldRuntime startContainer(String contextId, Environment environment, Dep
         if (!registry.contains(ProxyServices.class)) {
             registry.add(ProxyServices.class, new SimpleProxyServices());
         }
+        // check if ProxyServices implementation supports class defining on integrator side
+        if (!registry.get(ProxyServices.class).supportsClassDefiningAndLoading()) {
```

Comment:
We should also add  a static `AtomicBoolean` based on which we will invoke the method (in addition to integrators implementing new SPI). That will prevent the code from being executed multiple time as `WeldStartup` is executed once _per deployment_.